File: Ruby/Civitas/lib/casilla_impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaImpuesto < Casilla
    def initialize(name, cantidad)
      super(name)
      @importe=cantidad
    end

    # Método sobrescrito
    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        todos[actual].paga_impuesto(@importe)
      end
    end
  
    def to_s
       "\n   DESCRIPCIÓN CASILLA: 
        \n\t Nombre: #{nombre}
        \n\t Tipo: IMPUESTO
        \n\t Precio a pagar: #{@importe}
        "
    end
    
  end
end

File: Java/P5/Civitas/src/civitas/CasillaImpuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author arturo
 */
public class CasillaImpuesto extends Casilla{
    private float cantidad;
    
    CasillaImpuesto(float cantidad, String nombre){
        super(nombre);
        this.cantidad=cantidad;
    }
    
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(cantidad);
        }
    }
}
